packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -19,11 +19,13 @@
 import { injectable } from 'inversify';
 import { Disposable, DisposableCollection, Emitter, Event } from '../../common';
 import { Deferred } from '../../common/promise-util';
+import { PreferenceScope, PreferenceChange } from './preference-service';
 
 @injectable()
 export class PreferenceProvider implements Disposable {
```

Comment:
Paul suggested to make this class abstract.

Reply:
Only to have `getPreferences` as abstract as well because I didn't see why we would have providers that return nothing in any situations, but I don't have a strong opinion here :)